hash/hasher_test.go

```diff
@@ -242,4 +242,29 @@ func TestCompare(t *testing.T) {
 	assert.Error(t, hash.Compare(context.Background(), []byte("test"), []byte("$scrypt$ln=16385,r=8,p=1$2npRo7P03Mt8keSoMbyD/tKFWyUzjiQf2svUaNDSrhA=$MiCzNcIplSMqSBrm4HckjYqYhaVPPjTARTzwB1cVNYE=")))
 	assert.Error(t, hash.Compare(context.Background(), []byte("tesu"), []byte("$scrypt$ln=16384,r=8,p=1$2npRo7P03Mt8keSoMbyD/tKFWyUzjiQf2svUaNDSrhA=$MiCzNcIplSMqSBrm4HckjYqYhaVPPjTARTzwB1cVNYE=")))
 	assert.Error(t, hash.Compare(context.Background(), []byte("tesu"), []byte("$scrypt$ln=abc,r=8,p=1$2npRo7P03Mt8keSoMbyD/tKFWyUzjiQf2svUaNDSrhA=$MiCzNcIplSMqSBrm4HckjYqYhaVPPjTARTzwB1cVNYE=")))
+
+	assert.Nil(t, hash.Compare(context.Background(), []byte("test"), []byte("$md5$v=1$098f6bcd4621d373cade4e832627b4f6")))
+	assert.Nil(t, hash.CompareMD5(context.Background(), []byte("test"), []byte("$md5$v=1$098f6bcd4621d373cade4e832627b4f6")))
+	assert.Nil(t, hash.CompareMD5(context.Background(), []byte("test"), []byte("$md5$v=1$098F6BCD4621D373CADE4E832627B4F6")))
+	assert.Error(t, hash.Compare(context.Background(), []byte("test"), []byte("$md5$v=1$5a105e8b9d40e1329780d62ea2265d8a")))
```

Comment:
```suggestion
	assert.Error(t, hash.Compare(context.Background(), []byte("test"), []byte("$md5$v=1$5a105e8b9d40e1329780d62ea2265d8a")))
	assert.Error(t, hash.Compare(context.Background(), []byte("test"), []byte("$md5$v=1$ZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZZ")))

```

Reply:
To improve test coverage ;)

Reply:
Oh thanks for that :) I'm going to merge them.